tests/Unit/Evolution/Systems/ScalarAdvection/Test_TimeDerivativeTerms.cpp

```diff
@@ -0,0 +1,58 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/ScalarAdvection/Tags.hpp"
+#include "Evolution/Systems/ScalarAdvection/TimeDerivativeTerms.hpp"
+#include "Framework/CheckWithRandomValues.hpp"
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+
+template <size_t Dim>
+void test_time_derivative(const gsl::not_null<std::mt19937*> generator,
+                          const size_t number_of_pts) {
+  // create datavectors to store computed values of dudt and flux
+  Scalar<DataVector> dudt{number_of_pts, 0.0};
+  tnsr::I<DataVector, Dim, Frame::Inertial> flux(number_of_pts);
+
+  // generate random u and velocity field
+  std::uniform_real_distribution<> distribution(-1.0, 1.0);
+  const auto u = make_with_random_values<Scalar<DataVector>>(
+      generator, make_not_null(&distribution), dudt);
+  const auto velocity =
+      make_with_random_values<tnsr::I<DataVector, Dim, Frame::Inertial>>(
+          generator, make_not_null(&distribution), dudt);
+
+  // expected values of dudt and flux. note that dudt should be zero since we
+  // have no source terms
+  const Scalar<DataVector> dudt_expected{number_of_pts, 0.0};
+  const tnsr::I<DataVector, Dim, Frame::Inertial> flux_expected{
+      pypp::call<tnsr::I<DataVector, Dim, Frame::Inertial>>(
+          "TestFunctions", "compute_flux", u, velocity)};
+
+  // check values
+  ScalarAdvection::TimeDerivativeTerms<Dim>::apply(&dudt, &flux, u, velocity);
+  CHECK_ITERABLE_APPROX(dudt, dudt_expected);
```

Comment:
dudt is not set to anything by the `apply` function, so it should really just be whatever you passed in... I think it's fine to check it's 0 but only if you clarify why: that it hasn't been changed. The 0 itself is arbitrary. I would add a comment here clarifying this, and also the note on line 36 that says it should be 0, and also rename `dudt_expected` to `dudt_original` to help clarify you just want it to be unchanged.

Reply:
@fmahebert Thanks! I tried a bit more clear way in this fixup commit.

Reply:
@fmahebert Thanks! I tried a bit more clear way in the fixup commit.